### PR TITLE
feat(ctb): make MIPS oracle addrs immutable

### PR DESCRIPTION
**Pull request by Inphi**
PreimageOracle address in MIPS.sol is qualified as immutable for extra safety. An explicit getter function is provided to maintain external ABI compat.

### CODE REVIEW

File: packages/contracts-bedrock/src/cannon/MIPS.sol [hunk 1]
```diff
@@ -54,11 +54,15 @@ contract MIPS {
     uint32 constant EINVAL = 0x16;
 
     /// @notice The preimage oracle contract.
-    IPreimageOracle public oracle;
+    IPreimageOracle internal immutable ORACLE;
```
Comment (tynes) [inline packages/contracts-bedrock/src/cannon/MIPS.sol:56]: This would imply that all chains would use the same mips contract

Comment (Inphi) [reply packages/contracts-bedrock/src/cannon/MIPS.sol:56]: The consequence, IIUC, is mips contracts across all chains have the same codehash? This is fine though.

Comment (Inphi) [reply packages/contracts-bedrock/src/cannon/MIPS.sol:56]: The MIPS contract should be completely stateless, operating as a _pure_ function. It works the same way across all networks. Having an immutable oracle baked into the contract gets us there.

File: packages/contracts-bedrock/src/cannon/MIPS.sol [hunk 2]
```diff
@@ -54,11 +54,15 @@ contract MIPS {
...
 
     /// @param _oracle The address of the preimage oracle contract.
     constructor(IPreimageOracle _oracle) {
-        oracle = _oracle;
+        ORACLE = _oracle;
+    }
+
```
Comment (tynes) [inline packages/contracts-bedrock/src/cannon/MIPS.sol:62]: needs natspec

**Comment by tynes** [comment]
Looks good to me once the natspec is added!

File: packages/contracts-bedrock/src/cannon/MIPS.sol (hunk 2 above)
Comment (Inphi) [reply packages/contracts-bedrock/src/cannon/MIPS.sol:62]: Yup. Natspec added.

**Comment by refcell** [comment]
Just needs a bindings fix and rebase on develop for flaky tests (?)

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
@mergifyio refresh

**Pull request merged**